int8 speed_0
int8 speed_1
int8 speed_2
int8 speed_3